src/macro_support.rs

```diff
@@ -0,0 +1,39 @@
+use core::cmp::Ordering;
+
+#[inline]
+fn flatten<T>(value: (usize, &Option<T>)) -> Option<(usize, &T)> {
+    match value {
+        (index, Some(value)) => Some((index, value)),
+        _ => None,
+    }
+}
+
+/// `partial_cmp` implementation over iterators which ensures that storage
+/// ordering between `None` and `Some` is handled in a reasonable manner.
+#[doc(hidden)]
```

Comment:
Don't think these need to be `doc(hidden)` since the module is. Might want to copy the following private API warning from option_bucket to this module as well:

```
//! # PRIVATE API
//!
//! This API is private, for use only in the `derive(Key)` macro.
//! Usage for other purposes is not supported, and this API will
//! not abide by Semver stability guarantees.
```

Reply:
Added the warning instead. Cheers!